src/phase_change/PhaseChangeHierarchyIntegrator.cpp

```diff
@@ -0,0 +1,1201 @@
+        if (input_db->keyExists("T_convective_difference_form"))
+            d_T_convective_difference_form =
+                string_to_enum<ConvectiveDifferencingType>(input_db->getString("T_convective_difference_form"));
+        else if (input_db->keyExists("T_convective_difference_type"))
+            d_T_convective_difference_form =
+                string_to_enum<ConvectiveDifferencingType>(input_db->getString("T_convective_difference_type"));
+        else if (input_db->keyExists("default_T_convective_difference_form"))
+            d_T_convective_difference_form =
+                string_to_enum<ConvectiveDifferencingType>(input_db->getString("default_T_convective_difference_form"));
+        else if (input_db->keyExists("default_T_convective_difference_type"))
+            d_T_convective_difference_form =
+                string_to_enum<ConvectiveDifferencingType>(input_db->getString("default_T_convective_difference_type"));
+        if (input_db->keyExists("T_convective_time_stepping_type"))
+            d_T_convective_time_stepping_type =
+                string_to_enum<TimeSteppingType>(input_db->getString("T_convective_time_stepping_type"));
+    }
+}
+
+void
+PhaseChangeHierarchyIntegrator::getFromRestart()
+{
+    Pointer<Database> restart_db = RestartManager::getManager()->getRootDatabase();
+    Pointer<Database> db;
+    if (restart_db->isDatabase(d_object_name))
+    {
+        db = restart_db->getDatabase(d_object_name);
+    }
+    else
+    {
+        TBOX_ERROR(d_object_name << ":  Restart database corresponding to " << d_object_name
+                                 << " not found in restart file." << std::endl);
+    }
+    int ver = db->getInteger("PC_HIERARCHY_INTEGRATOR_VERSION");
+    if (ver != PC_HIERARCHY_INTEGRATOR_VERSION)
+    {
+        TBOX_ERROR(d_object_name << ":  Restart file version different than class version." << std::endl);
+    }
+
+    d_T_diffusion_time_stepping_type =
+        string_to_enum<TimeSteppingType>(db->getString("d_T_diffusion_time_stepping_type"));
+    d_T_convective_time_stepping_type =
+        string_to_enum<TimeSteppingType>(db->getString("d_T_convective_time_stepping_type"));
+    d_T_init_convective_time_stepping_type =
+        string_to_enum<TimeSteppingType>(db->getString("d_T_init_convective_time_stepping_type"));
+    d_T_convective_difference_form =
+        string_to_enum<ConvectiveDifferencingType>(db->getString("d_T_convective_difference_form"));
+    d_T_convective_op_type = db->getString("d_T_convective_op_type");
+    d_T_convective_op_input_db = db->getDatabase("d_T_convective_op_db");
+
+    d_latent_heat = db->getDouble("d_latent_heat");
+    d_rho_liquid = db->getDouble("d_rho_liquid");
+    d_rho_solid = db->getDouble("d_rho_solid");
+    d_T_melt = db->getDouble("d_T_melt");
+    d_solve_mass_conservation = db->getBool("d_solve_mass_conservation");
+    d_output_Div_U_F = db->getBool("d_output_Div_U_F");
+    d_output_temp_k = db->getBool("d_output_temp_k");
```

Comment:
We do this in a small number of places elsewhere in the library but we shouldn't - the `d_` is to indicate that something is a member variable. It shouldn't be in the input database.

Reply:
Thank you. I have corrected this everywhere.